src/content/app/tools/blast/state/blast-form/blastFormSlice.ts

```diff
@@ -97,16 +101,109 @@ const getDatabaseSequenceType = ({
   return config.database_sequence_types[database];
 };
 
+/**
+ * To be run for changes of sequences, sequence type, or database
+ */
+const autoUpdateSettings = (
+  state: BlastFormState,
+  config: BlastSettingsConfig
+) => {
+  if (
+    state.sequences.length &&
+    state.settings.sequenceSelectionMode === 'automatic'
+  ) {
+    const firstSequence = state.sequences[0].value;
+    const sequenceType = guessSequenceType(firstSequence); // FIXME — should this only be done if there were no sequences before?
```

Comment:
is the comment still valid? Because we still want to guess the sequence type if lets say the user either change the existing one or paste a different sequence?